ldap/servers/plugins/sync/sync_util.c

```diff
@@ -625,16 +675,21 @@ sync_cookie_parse(char *cookie, bool *cookie_refresh)
 
     sc = (Sync_Cookie *)slapi_ch_calloc(1, sizeof(Sync_Cookie));
     if (strncmp(cookie, "rid=", 4) == 0) {
+        if (*allow_openldap_compat != PR_TRUE) {
```

Comment:
It looks that the detection of openldap cookie is a bit fragile.
If the cookie is '_host:389#uid=foo,nsServerID=123,dc=com:dc=com:(objectclass=*)#4321_' then '_nsServerID_' will erroneously match the condition.

Reply:
Good catch, I wonder if there is some way to bound the strncmp here ...

Reply:
Re-reading, strncmp will only compare the literal first for characters, so your nsServerid=123 example won't work because it will actually compary `rid=` to `host`.